src/python/pants/backend/python/rules/pex.py

```diff
@@ -50,6 +58,14 @@ def create_requirements_pex(
     argv.extend(["--entry-point", request.entry_point])
   argv.extend(interpreter_constraint_args + list(request.requirements))
 
+  source_dir_name = 'source_files'
+
+  argv.append(f'--sources-directory={source_dir_name}')
```

Comment:
No need to append this arg if there are no sources.

Reply:
True, but in that case this directory will be empty and pex won't pick up any files, so I think it's cleaner to avoid a branch in the code.

Reply:
Agreed with Stu that this would be good to put under an if branch. While it's true that it will still work as is, that is not immediately clear to a casual observer. The extra LOC seems worth it to avoid the possible confusion.